share/gpodder/extensions/youtube-dl.py

```diff
@@ -411,16 +415,30 @@ def fetch_channel(self, channel, max_episodes=0):
             return self.refresh(url, channel.url, max_episodes)
         return None
 
+    def is_supported_url(self, url):
+        if self.regex_cache[0].match(url) is not None:
+            return True
+        for r in self.regex_cache[1:]:
+            if r.match(url) is not None:
+                self.regex_cache.remove(r)
+                self.regex_cache.insert(0, r)
+                return True
+        with youtube_dl.YoutubeDL(self._ydl_opts) as ydl:
+            for ie in ydl._ies:
+                if ie.suitable(url) and ie.ie_key() not in self.ie_blacklist:
+                    self.regex_cache.insert(0, ie._VALID_URL_RE)
+                    return True
+        return False
+
     def custom_downloader(self, unused_config, episode):
         """
         called from registry.custom_downloader.resolve
         """
         if not self.force and not self.my_config.manage_downloads:
             return None
-        if re.match(r'''https://www.youtube.com/watch\?v=.+''', episode.url):
+        if self.is_supported_url(episode.url):
```

Comment:
I'm happy with not checking `episode.link` as long as you tested with and without youtube_dl managing subscriptions.

Reply:
Ok. It's still a separate commit in the current patch stack, but can be squashed if necessary.